docs/product/explore/session-replay/mobile/performance-overhead.mdx

```diff
@@ -0,0 +1,57 @@
+  - When sending a replay segment, the screenshots are further compressed into an `mp4` video file using `h.264` compression with a 75 kbps bitrate (the default behavior). This ensures a small payload size while preserving a perceivable replay quality. We observed nearly a 5x reduction in size when compressing five screenshots into a video segment.
+  - The video segment, alogn with other debugging information (such as breadcrumbs) is gzip-compressed on the client before transmission over HTTP. This compression is performed on a background thread, ensuring the application's main thread remains unaffected.
+  - To minimize disk space usage, screenshots and video segment files are deleted from the disk upon successful upload.
+- Sentry’s event ingestion infrastructure uses [distributed Points-of-Presence (PoPs)](https://blog.sentry.io/2022/03/31/sentry-points-of-presence-how-we-built-a-distributed-ingestion/) which place ingestion servers around the world and close to your users. When a Session Replay event is transmitted, the user’s device connects and transmits the event payload to the closest PoP available for their region. This greatly reduces end-to-end latency and minimizes the amount of networking overhead placed on the browser.
+- The SDK is built to gracefully downgrade when needed. In the event of sending too many replay events and hitting Sentry's rate limit, the SDK will disable recording to make sure the rate limit is respected. Once the rate limit is lifted, the SDK will continue recording.
+
+## Factors in Performance Overhead
+
+While the performance overhead of our mobile replay SDKs will be imperceptible to end-users most of the time, it can vary based on the complexity of your application. If an application has screens with a deeply-nested view tree, there will be higher overhead compared to simpler (e.g. Settings) screens.
+
+If masking is enabled (the default behavior), network overhead is further reduced because the screenshots have fewer pixels, resulting in smaller file sizes.
+
+There are different stages in the lifecycle of a mobile application, each with different performance characteristics. Here is a list of stages in the lifecycle, and the most important performance metrics to be aware of.
+
+### App Start
+
+There is virtually no overhead from the SDK during application startup. It performs only a few lightweight tasks, such as registering the necessary listeners for system events and scheduling a background thread for capturing.
+
+### Main Thread
+
+The SDK performs only the bare minimum on the main thread to capture a new frame for the replay (a screenshot and a view hierarchy snapshot) before immediately switching to a background thread for post-processing tasks. These tasks include masking screenshots, storing them in the disk cache, or encoding a video segment. This process occurs at most once per secondk, and it can be even less frequent depending on the frame rate setting.
```

Comment:
```suggestion
The SDK performs only the bare minimum on the main thread to capture a new frame for the replay (a screenshot and a view hierarchy snapshot) before immediately switching to a background thread for post-processing tasks. These tasks include masking screenshots, storing them in the disk cache, or encoding a video segment. This process occurs at most once per second, and it can be even less frequent depending on the frame rate setting.
```

Reply:
is this frame rate setting configurable? worth mentioning if it is and links to learn how you can configure it

Reply:
it's configurable but only internally for now, we may expose it in the future if there's a request